core/include/userver/engine/io/common.hpp

```diff
@@ -51,6 +52,12 @@ class ReadableBase {
   impl::ContextAccessor* ca_{nullptr};
 };
 
+class NonblockingReadableBase {
+ public:
+  // Either receives len bytes or returns -1 AND EAGAIN/EWOULDBLOCK
```

Comment:
`// Receives 0 or more bytes from the stream without waiting for any bytes to come`

Do not show low level POSIX stuff to framework users :)

Reply:
We could've dropped this comment, but leave -1 + EWOULDBLOCK/EAGAIN as implementation detail. What do you think?

Reply:
Upd. as we'll drop this _new_ iface => I think, the question is no longer relevant.